analysis/monkmistweaver/src/modules/core/HotTrackerMW.tsx

```diff
@@ -23,37 +23,39 @@ class HotTrackerMW extends HotTracker {
     this.upwellingActive = selectedCombatant.hasTalent(SPELLS.UPWELLING_TALENT.id);
   }
 
-  calculateMaxDuration(event: ApplyBuffEvent & { selectedCombatant: Combatant }) {
+  calculateMaxDuration(event: ApplyBuffEvent, combatant: Combatant) {
```

Comment:
tacking an extra field onto the existing passed Event made no sense and is very awkward in TypeScript. I instead just made the combatant an extra parameter.